### ISSUE TITLE
Cannot switch on master stream source

### ISSUE THREAD

**Issue by mikeopensauce** [LibreTime/libretime#725]
Hi all.
I have installed Libretime (I can't see which version. Can anyone tell me how to find that out?) on a Ubuntu 16.04 VPS and everything works fine except for the following.
I have setup and connected a master streaming source using BUTT. When I connect and send audio, BUTT says it is connected and LibreTime says there is a master source (Orange link shows next to "Master Source". See screenshot) but I cannot switch the stream on. Clicking on the switch at either end makes no difference as it doesn't move at all.
I have tried setting up Auto Switch on but that also made no difference.
Is anyone able to help suggest anything or point me in the right direction?
Man thanks
Mike

![screenshot_2019-01-31 rosetta radio - libretime](https://user-images.githubusercontent.com/7012099/52044702-a9e1f700-253a-11e9-86d3-a20a149f552f.png)

**Comment by mikeopensauce** [comment]
Further to this, I have now found the log in zendphp.log and when I click on the button to switch on the source I get the following error messages.
`2019-01-31T09:40:44+00:00 ERR (3): stream.rosettaradio.co.uk [ErrorController.php:26 - errorAction()] - ACCESS_REFUSED - Login was refused using authentication mechanism AMQPLAIN. For details see the broker logfile.
2019-01-31T09:40:44+00:00 ERR (3): stream.rosettaradio.co.uk [ErrorController.php:27 - errorAction()] - #0 [internal function]: PhpAmqpLib\Connection\AbstractConnection->connection_close(Object(PhpAmqpLib\Wire\AMQPReader))
#1 /usr/share/airtime/php/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Channel/AbstractChannel.php(199): call_user_func(Array, Object(PhpAmqpLib\Wire\AMQPReader))
#2 /usr/share/airtime/php/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Channel/AbstractChannel.php(341): PhpAmqpLib\Channel\AbstractChannel->dispatch('10,50', '\x01\x93oACCESS_REFUS...', NULL)
#3 /usr/share/airtime/php/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Connection/AbstractConnection.php(212): PhpAmqpLib\Channel\AbstractChannel->wait(Array)
#4 /usr/share/airtime/php/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Connection/AbstractConnection.php(180): PhpAmqpLib\Connection\AbstractConnection->connect()
#5 /usr/share/airtime/php/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Connection/AMQPStreamConnection.php(59): PhpAmqpLib\Connection\AbstractConnection->__construct('airtime', 'airtime', '/airtime', false, 'AMQPLAIN', NULL, 'en_US', Object(PhpAmqpLib\Wire\IO\StreamIO), 0)
#6 /usr/share/airtime/php/airtime_mvc/application/models/RabbitMq.php(23): PhpAmqpLib\Connection\AMQPStreamConnection->__construct('127.0.0.1', '5672', 'airtime', 'airtime', '/airtime')
#7 /usr/share/airtime/php/airtime_mvc/application/models/RabbitMq.php(50): Application_Model_RabbitMq::sendMessage('airtime-pypo', 'direct', true, '{"sourcename":"...')
#8 /usr/share/airtime/php/airtime_mvc/application/controllers/DashboardController.php(63): Application_Model_RabbitMq::SendMessageToPypo('switch_source', Array)
#9 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Controller/Action.php(516): DashboardController->switchSourceAction()
#10 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Controller/Dispatcher/Standard.php(308): Zend_Controller_Action->dispatch('switchSourceAct...')
#11 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Controller/Front.php(954): Zend_Controller_Dispatcher_Standard->dispatch(Object(Zend_Controller_Request_Http), Object(Zend_Controller_Response_Http))
#12 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Application/Bootstrap/Bootstrap.php(105): Zend_Controller_Front->dispatch()
#13 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Application.php(384): Zend_Application_Bootstrap_Bootstrap->run()
#14 /usr/share/airtime/php/airtime_mvc/application/airtime-boot.php(84): Zend_Application->run()
#15 /usr/share/airtime/php/airtime_mvc/public/index.php(68): require_once('/usr/share/airt...')
#16 {main}
2019-01-31T09:40:44+00:00 ERR (3): stream.rosettaradio.co.uk [ErrorController.php:54 - errorAction()] - An internal application error has occurred.: PhpAmqpLib\Exception\AMQPProtocolConnectionException: ACCESS_REFUSED - Login was refused using authentication mechanism AMQPLAIN. For details see the broker logfile. in /usr/share/airtime/php/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Connection/AbstractConnection.php:661
Stack trace:
#0 [internal function]: PhpAmqpLib\Connection\AbstractConnection->connection_close(Object(PhpAmqpLib\Wire\AMQPReader))
#1 /usr/share/airtime/php/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Channel/AbstractChannel.php(199): call_user_func(Array, Object(PhpAmqpLib\Wire\AMQPReader))
#2 /usr/share/airtime/php/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Channel/AbstractChannel.php(341): PhpAmqpLib\Channel\AbstractChannel->dispatch('10,50', '\x01\x93oACCESS_REFUS...', NULL)
#3 /usr/share/airtime/php/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Connection/AbstractConnection.php(212): PhpAmqpLib\Channel\AbstractChannel->wait(Array)
#4 /usr/share/airtime/php/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Connection/AbstractConnection.php(180): PhpAmqpLib\Connection\AbstractConnection->connect()
#5 /usr/share/airtime/php/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Connection/AMQPStreamConnection.php(59): PhpAmqpLib\Connection\AbstractConnection->__construct('airtime', 'airtime', '/airtime', false, 'AMQPLAIN', NULL, 'en_US', Object(PhpAmqpLib\Wire\IO\StreamIO), 0)
#6 /usr/share/airtime/php/airtime_mvc/application/models/RabbitMq.php(23): PhpAmqpLib\Connection\AMQPStreamConnection->__construct('127.0.0.1', '5672', 'airtime', 'airtime', '/airtime')
#7 /usr/share/airtime/php/airtime_mvc/application/models/RabbitMq.php(50): Application_Model_RabbitMq::sendMessage('airtime-pypo', 'direct', true, '{"sourcename":"...')
#8 /usr/share/airtime/php/airtime_mvc/application/controllers/DashboardController.php(63): Application_Model_RabbitMq::SendMessageToPypo('switch_source', Array)
#9 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Controller/Action.php(516): DashboardController->switchSourceAction()
#10 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Controller/Dispatcher/Standard.php(308): Zend_Controller_Action->dispatch('switchSourceAct...')
#11 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Controller/Front.php(954): Zend_Controller_Dispatcher_Standard->dispatch(Object(Zend_Controller_Request_Http), Object(Zend_Controller_Response_Http))
#12 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Application/Bootstrap/Bootstrap.php(105): Zend_Controller_Front->dispatch()
#13 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Application.php(384): Zend_Application_Bootstrap_Bootstrap->run()
#14 /usr/share/airtime/php/airtime_mvc/application/airtime-boot.php(84): Zend_Application->run()
#15 /usr/share/airtime/php/airtime_mvc/public/index.php(68): require_once('/usr/share/airt...')
#16 {main}
`

**Comment by mikeopensauce** [comment]
And I have now found the version I installed!
3.0.0-alpha.5 installed from the tar.gz on github

**Comment by mikeopensauce** [comment]
I have solved this myself (which is a miracle!).
For anyone else, it was solved by the advice given here on [Sourcefabric](https://sourcefabric.booktype.pro/airtime-25-for-broadcasters/troubleshooting/) which explained that if the server name has been changed, this can break RabbitMQ. The instructions to fix it are copied below in case they go missing.

**RabbitMQ hostname changes**

If the Airtime logs indicate failures to connect to the RabbitMQ server, such as:

2013-10-31 08:21:11,255 ERROR - [pypomessagehandler.py : main() : line 
99] - Error connecting to RabbitMQ Server. Trying again in few seconds

2013-10-31 08:21:11,255 ERROR - [pypomessagehandler.py : main() : line 99] - Error connecting to RabbitMQ Server. Trying again in few seconds - See more at: http://forum.sourcefabric.org/discussion/16050/#sthash.W8OJrNFm.dpuf

but the RabbitMQ server is running normally, this error might be due to a change in the server's hostname since Airtime installation. Directory names under /var/lib/rabbitmq/mnesia/ indicate that RabbitMQ's database files are organised according to the hostname of the server, for example:

rabbit@airtime

where the hostname is airtime.example.com. If the hostname has changed, it may be necessary to reconfigure RabbitMQ manually, as follows:

1. Delete the files in /var/lib/rabbitmq/mnesia/

sudo rm -r /var/lib/rabbitmq/mnesia/*

2. Restart RabbitMQ:

sudo invoke-rc.d rabbitmq-server restart

3. Enter the following commands to set up authentication and grant permissions. The rabbitmqctl add_user command requires the RabbitMQ password from the /etc/airtime/airtime.conf file as an argument. The rabbitmqctl set_permissions command should be entered on one line, with the list of Airtime services repeated three times:

rabbitmqctl add_vhost /airtime
rabbitmqctl add_user airtime XXXXXXXXXXXXXXXXXXXX 
rabbitmqctl set_permissions -p /airtime airtime 
   "airtime-pypo|pypo-fetch|airtime-media-monitor|media-monitor"
   "airtime-pypo|pypo-fetch|airtime-media-monitor|media-monitor"
   "airtime-pypo|pypo-fetch|airtime-media-monitor|media-monitor"
